core/ui/compose/designsystem/src/test/kotlin/app/k9mail/core/ui/compose/designsystem/atom/textfield/PasswordTextFieldOutlinedKtTest.kt

```diff
@@ -0,0 +1,165 @@
+        ).performClick()
+
+        onNodeWithText(VALUE).assertIsDisplayed()
+    }
+
+    @Test
+    fun `should not toggle password visibility when disabled`() = runComposeTest {
+        setContent {
+            PasswordTextFieldOutlined(
+                value = VALUE,
+                onValueChange = ON_VALUE_CHANGE,
+                enabled = false,
+            )
+        }
+
+        val showPasswordNode = onNodeWithContentDescription(
+            getString(R.string.core_ui_designsystem_accessibility_show_password),
+        )
+        val hidePasswordNode = onNodeWithContentDescription(
+            getString(R.string.core_ui_designsystem_accessibility_hide_password),
+        )
+
+        // Check initial state
+        showPasswordNode.assertIsDisplayed()
+        hidePasswordNode.assertDoesNotExist()
+
+        // Toggle password visibility
+        showPasswordNode.performClick()
+
+        // Check toggled state
+        showPasswordNode.assertIsDisplayed()
+        hidePasswordNode.assertDoesNotExist()
+        onNodeWithText(VALUE).assertDoesNotExist()
+    }
+
+    @Test
+    fun `should call onValueChange when value changes`() = runComposeTest {
+        var value = VALUE
+        setContent {
+            PasswordTextFieldOutlined(
+                value = value,
+                onValueChange = { value = it },
+                modifier = Modifier.testTag(TEST_TAG),
+            )
+        }
+
+        onNodeWithTag(TEST_TAG).performClick()
+        onNodeWithTag(TEST_TAG).performTextInput(" + added text")
+
+        assertThat(value).isEqualTo("$VALUE + added text")
+    }
+
+    @Test
+    fun `should be enabled by default`() = runComposeTest {
+        setContent {
+            PasswordTextFieldOutlined(
+                value = VALUE,
+                onValueChange = ON_VALUE_CHANGE,
+                modifier = Modifier.testTag(TEST_TAG),
+            )
+        }
+
+        onNodeWithTag(TEST_TAG).assertIsEnabled()
+    }
+
+    @Test
+    fun `should be disabled when enabled is false`() = runComposeTest {
+        setContent {
+            PasswordTextFieldOutlined(
+                value = VALUE,
+                onValueChange = ON_VALUE_CHANGE,
+                modifier = Modifier.testTag(TEST_TAG),
+                enabled = false,
+            )
+        }
+
+        onNodeWithTag(TEST_TAG).assertIsNotEnabled()
+    }
+
+    @Test
+    fun `should show label when label is not null`() = runComposeTest {
+        setContent {
+            PasswordTextFieldOutlined(
+                value = VALUE,
+                onValueChange = ON_VALUE_CHANGE,
+                label = LABEL,
+            )
+        }
+
+        onNodeWithText(LABEL).assertIsDisplayed()
+    }
+
+    private companion object {
+        const val TEST_TAG = "PasswordTextFieldOutlined"
+        const val VALUE = "Input text"
+        const val LABEL = "Label"
+        val ON_VALUE_CHANGE: (String) -> Unit = {}
```

Comment:
I like adding constants like these as private top level properties in tests. That also allows you to put them at the top of the file.

See e.g. https://github.com/thundernest/k-9/blob/730825bce30a6eef522395d1b4fe3edae78da206/app/ui/legacy/src/test/java/com/fsck/k9/ui/messageview/MessageViewRecipientFormatterTest.kt#L15-L16